prime-router/docs/api/README.md

```diff
@@ -1,16 +1,168 @@
+by ReportStream and provided to the client, that is encoded as a JSON Web-Token. This token is submitted to the server
+in order to request a bearer token. This token will then be used to authenticate further requests. It expires after 5
+minutes so it must be periodically refreshed using a separate API call.
 
+Provide the client ID / secret and select a scope in the Authorize menu and Swagger will automate the above OAuth process.
 
+#### Server-to-server
+
+###### Background
+This method is based on best practices outlined in the SMART on FHIR. Similar to Okta above, this method follows the
+principles of OAuth2, similar to the Okta method above. A `client secret`, `client ID`, and `scope` is encoded as a
+JSON Web-Token and sent to the server in a header value. The server then responds with a bearer token the client can use
+to authenticate and authorize further requests. Similar to Okta, this token will expire every 5 minutes and must be
+refreshed with a separate API call.
+
+###### Generating a client key
+Unlike Okta, the client secret must be generated by the client and provided to ReportStream. Instructions and a related helper python
+script is provided in this repo (examples/generate-jwt-python). See the README in that directory for instructions,
+summarized here:
+```bash
+openssl genrsa -out my-rsa-private-key 2048
+openssl rsa -in my-rsa-private-key -outform PEM -pubout -out my-rsa-public-key.pem
+```
+
+###### Encoding credentials as a JWT to be sent to the server
+The initial JWT sent to the server must be created outside of the swagger environment. An example using a python script
+is provided in this repo (examples/generate-jwt-python). See the README in that directory for instructions, summarized
+here:
+1. Install python dependencies for generating JWT
+2. Set variables on lines 28-30 in the script (`my_client_id`, `my_kid`, and `my_rsa_keypair_file`)
+3. Run the script! The penultimate curl contains the JWT used to retrieve a bearer token. The final curl contains an
+   example using that bearer token.
+
+#### API Key (x-functions-key)
+
+This method utilizes a simple API key and a client ID that is mapped to a header value and sent with every request.
+It is created and stored in the Azure cloud per Azure function by ReportStream engineers and provided to the end user
+during onboarding.  Important to note that this method is deprecated due to security concerns and should only be used in
+edge cases where Okta or server-server auth is not possible. 
+
+## Developer Notes
+
+The openapi spec for ReportStream is generated from annotations in the code. This facilitates accurate, timely
+updates and ameliorates an otherwise lengthy, manual, repetitive, and error-prone process.
+
+### Swagger UI Customization and Maintenance
+Since the swagger files are generated from annotations in the code, any changes must be made to the annotations
```

Comment:
```suggestion
Since the OpenAPI spec files (in yaml format) are generated from annotations in the code, any changes must be made to the annotations
```